microsoft/kiota
# Feat: Ext/task5206/command based ttk integration in vs code extension

Partial fix for #5206

To test:
1. Follow steps specified in [debugging.md](https://github.com/microsoft/kiota/blob/main/vscode/microsoft-kiota/debugging.md#running-in-debug-mode)
2. On pressing F5 to open a new vscode debug window, run these next commands in that debug window
3. Create|open an md file e.g "test.md" in/from your preferred location and create a commandUri that enables you to pass requisite parameters. See sample attached. 
[test.md](https://github.com/user-attachments/files/16787882/test.md)

4. ctrl + click on the command uri to execute the command which is similar to doing 
```
vscode.commands.executeCommand(
            'kiota.openApiExplorer.searchOrOpenApiDescription',
            [
                {
                    "kind":"Plugin|Client",
                    "type":"ApiPlugin|ApiManifest|OpenAI", //Only useful when kind is plugin
                    "language": "csharp|python|go|typescript|java|php|ruby|swift|cli" //currently supported kiota languages. Only important when kind=Client
                    "name":"GithubRestPlugin", // or any other suitable name
                    "source":"ttk|<any other source name as string>"
                }
            ]   
        );
```
i.e. passing params as an array of a single object with given keys and value pairs
5. The command should skip predefined questions e.g what to generate, name and language as applicable but still generate client|plugin after you provide output folder
6. Next step of work may include auto populating folder path with temp folder when source=ttk for seamless integration. Intentionally puting the temp folder generation on hold until integration team agrees on way forward as agreed.
